### ISSUE TITLE
Mealplan week totals calculation not correct

### ISSUE THREAD

**Issue by crazyelectron-io** [grocy/grocy#808]
How is the totaling of a weeks meal plan is supposed to work? I've set the week to start on Saturday and with an empty meal plan week it still shows a total cost of EUR 43.39.

**Comment by berrnd** [comment]
Most probably your timezone settings between the used client and the server are different.

Currently all time related things use the local time, therefore the timezones must be the same. I don't know if it would be possible to use a different "start of week" for SQLite (where the week total costs are currently calculated)...

**Comment by berrnd** [comment]
References partly #214.

**Comment by berrnd** [comment]
Somehow in all SQL views (where needed) the `config.php` setting `CALENDAR_FIRST_DAY_OF_WEEK` and `MEAL_PLAN_FIRST_DAY_OF_WEEK` would be needed to be considered.

**Comment by jac4e** [comment]
I ran across this issue when setting up grocy and trying to find out why my meal plan ingredients were not being added to the shopping list properly.

The main issue seems to be sqlites STRFTIME functions %W formatter gives the week of the year with the week starting on monday, somewhat following ISO 8601 (I dont believe this is something that can be changed to allow custom start of weeks).

The only workaround I have found is to just set grocy's start of week to monday, eliminating any incongruities on the meal plan page. Which I would suggest Monday should be the only option currently even if some (like in #429) prefer starting their weeks on different days as it still would require flipping through multiple weeks to get all the missing products added to the shopping list.

Another option I have thought about is to remove the idea of mealplan-weeks from grocy's internals entirely and just focus on mealplan-dates, with costs and 'add to the shopping list' being generated from a given range of dates. This would solve this issue, more fully address #429 in regards to 'add to the shopping list', and potentially allow meal planning a month at a time.

**Comment by berrnd** [comment]
> The root issue seems to be sqlites STRFTIME functions %W formatter gives the week of the year with the week starting on monday, somewhat following ISO 8601

That's right and exactly what this here is about - all what you pointed out is nothing new, but thanks anyway.

And of course that's not impossible to be solved (even custom functions in SQLite would be possible to e.g. pass configuration options into it, [other related Grocy example](https://github.com/grocy/grocy/blob/edf973df00b327344c589536a3b67be45890196d/services/DatabaseService.php#L100-L104)) - we don't try to fly to Mars here, it's just about managing your fridge.
